src/llama_recipes/finetuning.py

```diff
@@ -250,6 +250,10 @@ def main(**kwargs):
             pin_memory=True,
             **val_dl_kwargs,
         )
+        if len(eval_dataloader) == 0:
```

Comment:
As this will usually happen in a distributed case, will this lead to the collective hanging until timeout?

Reply:
From my previous test, all the worker will directly hit `raise ValueError` statement and exit immediately without any hanging.